src/components/activity/ActivityDecomposition.svelte

```diff
@@ -14,24 +14,27 @@
   export let selectedSpanId: SpanId | null = null;
   export let spansMap: SpansMap = {};
   export let spanUtilityMaps: SpanUtilityMaps;
+  export let childPageSize: number = 25;
 
   const dispatch = createEventDispatcher<{
     select: number | null;
   }>();
 
   let childIds: SpanId[] = [];
   let span: Span | null = null;
+  let childLimit = childPageSize;
 
   $: span = rootSpanId !== null ? spansMap[rootSpanId] : null;
   $: isRoot = span ? !span.parent_id : true;
   $: type = span?.type || '';
-  $: childIds = span !== null ? spanUtilityMaps?.spanIdToChildIdsMap[span?.id] : [];
+  $: childIds = span !== null ? spanUtilityMaps?.spanIdToChildIdsMap[span?.id] || [] : [];
   $: hasChildren = childIds ? childIds.length > 0 : false;
   $: role = isRoot ? 'tree' : 'treeitem';
   $: nodeClass =
     'activity-decomposition-node activity-decomposition-' +
     (rootSpanId === selectedSpanId ? 'selected st-typography-medium' : 'unselected st-typography-body');
   $: buttonClass = 'st-button icon' + (!hasChildren ? ' st-button-no-hover' : '');
+  $: childIdsInView = childIds.slice(0, childLimit);
```

Comment:
I know it's valid, but I personally think that explicitly declaring variables and the types are a little clearer (albeit more verbose). I could be convinced to go with the "svelteness", but I wanted to get your take on it @AaronPlave.

Reply:
Good call, thought I had declared it! Will add that in.

Reply:
Actually that said, now that I look at this component there are a number of undeclared reactive variables like `hasChildren`, `nodeClass`, `buttonClass`, `type`, etc. Curious what the best practice is for when a reactive statement should or should not have a backing explicit declaration?

Reply:
So maybe we should just pre-declare with a `let` when it'd be useful to have that type, say when it is a more complex or not obvious type? Or would you prefer to have all of these have their own pre-declarations?

Reply:
I'd say maybe just always pre-declare the type so that it's easier to know what the intended type of the variable is. When it's derived, my concern is that there's still a chance that the implementation in the reactive statement returns something unexpected and then the type mismatch isn't caught immediately.